ports/zephyr/main.c

```diff
@@ -97,11 +98,17 @@ static void vfs_init(void) {
     int ret = 0;
 
     #ifdef CONFIG_DISK_DRIVER_SDMMC
+    #if KERNEL_VERSION_NUMBER >= ZEPHYR_VERSION(4, 0, 0)
+    #define DT_DRV_COMPAT zephyr_sdmmc_disk
+    mp_obj_t args[] = { mp_obj_new_str_from_cstr(DT_INST_PROP(0, disk_name)) };
+    #else
     mp_obj_t args[] = { mp_obj_new_str_from_cstr(CONFIG_SDMMC_VOLUME_NAME) };
+    #endif
     bdev = MP_OBJ_TYPE_GET_SLOT(&zephyr_disk_access_type, make_new)(&zephyr_disk_access_type, ARRAY_SIZE(args), 0, args);
     mount_point_str = "/sd";
     #elif defined(CONFIG_FLASH_MAP) && FIXED_PARTITION_EXISTS(storage_partition)
-    mp_obj_t args[] = { MP_OBJ_NEW_SMALL_INT(FIXED_PARTITION_ID(storage_partition)), MP_OBJ_NEW_SMALL_INT(4096) };
+    #define DT_DRV_COMPAT zephyr_flash_disk
+    mp_obj_t args[] = { MP_OBJ_NEW_SMALL_INT(FIXED_PARTITION_ID(storage_partition)), MP_OBJ_NEW_SMALL_INT(DT_INST_PROP(0, cache_size)) };
```

Comment:
This change (to retrieve `cache-size` from the device tree) now requires all boards that use this bit of code to have this defined.  And that seems to be something you have to now manually do.

Eg, with this PR the existing `nucleo_h743zi` board fails to build (using Zephyr v3.7.0):
```
zephyr/include/zephyr/devicetree.h:336:40: error: 'DT_N_INST_0_zephyr_flash_disk_P_cache_size' undeclared (first use in this function)
```

Reply:
Yes, you are right.  I do not have an overview of how many boards are affected, but I see that this is a problem.  Would it be ok, to check if such a property exists and if not use the hard coded 4 Kib instead?  I'll try to rework that patch in this way.

Reply:
> Would it be ok, to check if such a property exists and if not use the hard coded 4 Kib instead?

Yes, if that's possible, that would be good.

Reply:
I just force-pushed an update with this modification.  In order to let people know that the device tree should be updated, I inserted a corresponding #warning. I hope this is ok?

Also, I rewrote the device tree access not to use the #DRV_COMPAT macro that I used previously. I think things should be self-contained, even if that takes longer to express in code.

I am having a lot of problems getting the code formatting right :facepalm: